name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/SecuritiesChart.java

```diff
@@ -149,6 +162,40 @@ public void widgetSelected(SelectionEvent e)
         });
     }
 
+    private static final class JSColors
+    {
+        private static final int SIZE = 11;
+        private static final float STEP = 360.0f / (float) SIZE;
+
+        private static final float HUE = 262.3f;
+        private static final float SATURATION = 0.464f;
+        private static final float BRIGHTNESS = 0.886f;
+
+        public int[] byIndex(int colorIndex, float setSaturation)
+        {
+            float brightness = Math.min(1.0f, BRIGHTNESS + (0.05f * (colorIndex / SIZE)));
+            return ColorConversion.toRGB(ColorConversion.toHex((HUE + (STEP * colorIndex)) % 360f, setSaturation, brightness));
+        }
+    }
+
+    
+    private ILineSeries chartSeriesPainter (ILineSeries seriesIn, LocalDate[] dates, double[] values, int colorIndex, float setSaturation,
+                                            int lineWidth, LineStyle lineStyle, boolean enableArea, boolean visibleLegend)
+    { 
+        JSColors colors = new JSColors();
+        int[] rgbColor;
+        rgbColor = colors.byIndex(colorIndex, setSaturation);
+        if (lineWidth != 0) seriesIn.setLineWidth(lineWidth);
+        seriesIn.setLineStyle(lineStyle);
+        seriesIn.enableArea(enableArea);
+        seriesIn.setXDateSeries(TimelineChart.toJavaUtilDate(dates));
+        seriesIn.setYSeries(values);
+        seriesIn.setAntialias(SWT.ON);
+        if(setSaturation != 0) seriesIn.setLineColor(new Color(Display.getDefault(), new RGB(rgbColor[0], rgbColor[1], rgbColor[2])));
```

Comment:
Achtung. Color ist eine SWT Resource. Und die muss auch immer explizit abgeräumt werden (also nicht die übliche Garbage Collection). Ich habe in PP mir die Hilfsklasse ```Colors``` gebaut in der ich alle Farben, die sowieso wiederverwendet werden, gecacht werden. In diesem Fall kann man die verwenden (es ist ja nicht so das wir ständig neue Farben für das Chart verwenden).